### DISCUSSION TITLE
How can I morph additional content without deleting existing?

### DISCUSSION THREAD

**Discussion by manuelmeister** [alpinejs/alpine#3895] [1. Help]
In my html response loaded by the server, I have the html of additional children (it might contain already existing children), and want to add everything.

Currently the Morph plugin is similiar to a right join in database terms. However I would love to have a full outer join. I tried to set 
```js
removing(el, skip) {
    return skip();
},
```
but that did not work, as it has 6 elements on page load and when I load 6 more items and morph them, then they are just updated.

**Comment by ekwoka** [comment]
If you give them keys and do the skip, you'd get the result you want (well, matching keys would merge, but unique would be added)

But you could also just append the children where you want them...

and turn on lookahead.

**Reply by manuelmeister** [reply]
Strange, because I've set the keys and it got merged, even though it didn't match. This is how I've tried it:

$refs.itemcontainer.innerHTML
```html
<article data-key="askfjk" x-show="isVisible">
  …
</article>
<article data-key="732c4r"…
```

fetched content:
```html
<article data-key="abd32x" x-show="isVisible">
  …
</article>
<article data-key="3hf2ls"…
```

code:
```typescript
Alpine.morph(this.$refs.itemcontainer, `<div data-childrenonly>${fetchedContent}</div>`, {
    key(el) {
        return el.dataset.key;
    },
    removing(el, skip) {
        return skip();
    },
    updating(el, toEl, childrenOnly, skip) {
        if (
            el !== undefined &&
            el.dataset &&
            "childrenonly" in el.dataset &&
            el.dataset.childrenonly == ""
        ) {
            return childrenOnly();
        }
    },
    lookahead: true,
});
```
